posts/intake_esm_dask.md

```diff
@@ -0,0 +1,197 @@
+
+```python
+catalog_file = '/glade/u/home/kristenk/TTE_CESM-LE/krill-cesm-le/notebooks/data/glade-cesm1-le.json'
+variables = ['TEMP', 'diatC']
+
+experiments = ['20C', 'RCP85']
+stream = 'pop.h'
+
+col = intake.open_esm_datastore(catalog_file, sep=',')
+col
+```
+
+Now we search the collection for the ensemble members (unique `member_id`'s) that have a chlorophyll field (`diatChl`). This is necessary because the ocean biogeochemistry was corrupted in some members and the data were deleted.
+
+In this cell, `member_id` is a list of the ensemble members we want to operate on.
+
+```python
+col_sub = col.search(experiment=['20C'],
+                     stream='pop.h',
+                     variable=['diatChl'])
+
+member_id = list(col_sub.df.member_id.unique())
+print(member_id)
+```
+
+### Apply the search for data
+
+Specify a list of variables and perform a search. Under the hood, the `search` functionality uses [`pandas`](https://pandas.pydata.org/) data frames. We can view that frame here using the `.df` syntax.
+
+```python
+col_sub = col.search(
+    experiment=experiments,
+    stream=stream,
+    variable=variables,
+    member_id=member_id,
+    )
+
+print(col_sub)
+
+col_sub.df.head()
+```
+
+Use `.to_dataset_dict` to read in the datasets as a dictionary, specifying the chunk size. This is also a point where it is recommended you take a look at the `chunksize` for the dask array, ensuring that it is a reasonable number (~100-200 mb). Test it out on one of the variables (ex. `TEMP`)
+
+```python
+dsets = col_sub.to_dataset_dict(cdf_kwargs={'chunks': {'time':5}, 'decode_times': False})
+```
+
+## Apply an operation
+
+Apply a calculation. In this case, we are calculating the average temperature within the top 100 m of the ocean.
+
+```python
+def compute_TEMP_100m(ds):
+    """compute top 100m mean temperature"""
+
+    ds['TEMP_100m_mean'] = ds.TEMP.isel(z_t=slice(0,10)).mean(dim='z_t')
+    ds.TEMP_100m_mean.attrs = ds.TEMP.attrs
+    ds.TEMP_100m_mean.attrs['long_name'] = 'Mean temperature over top 100m'
+
+    return ds.drop(['TEMP'])
+```
+
+Now apply this function to the dataset(s)
+
+```python
+# compute top 100m temperature
+dsets2 = {key: compute_TEMP_100m(ds) for key, ds in dsets2.items()}
+print('computed top 100m temp')
+```
+
+### Combine the datasets
+
+Concatenate the datasets in time, i.e. 20C + RCP8.5 experiments.
+
+```python
+ordered_dsets_keys = ['ocn,20C,pop.h', 'ocn,RCP85,pop.h']
+#ordered_dsets_keys = ['ocn.20C.pop.h', 'ocn.RCP85.pop.h']
+ds = xr.concat(
+    [dsets2[exp] for exp in ordered_dsets_keys],
+    dim='time',
+    data_vars='minimal',
+    #compat='override' ## added this
+)
+time_encoding = dsets2[ordered_dsets_keys[0]].time.encoding
+```
+
+### Calculate an annual mean
+
+Create an annual mean for temperature, using a utilility function which deals with the times correctly.
+
+```python
+ds_ann = util.ann_mean(ds, time_bnds_varname='time_bound', time_centered=True)
```

Comment:
Add the function definition here? I know you refer to the repo above... 

Alternatively, add a link to the function in the repo.